src/utils/UpdateNotification.js

```diff
@@ -115,6 +115,28 @@ define(function (require, exports, module) {
             locale = locale.substring(0, 2);
         }
 
+         //AUTOUPDATE_PRERELEASE_BEGIN
+        // The following code is needed for supporting Auto Update in prerelease,
+        //and will be removed eventually for stable releases
+        {
+            var jsonLocale = locale;
+            if (locale && locale.toLowerCase().startsWith("de")) {
+                jsonLocale = "de";
+            } else if (locale && locale.toLowerCase().startsWith("es")) {
+                jsonLocale = "es";
+            } else if (locale && locale.toLowerCase().startsWith("fr")) {
+                jsonLocale = "fr";
+            } else if (locale && locale.toLowerCase().startsWith("ja")) {
+                jsonLocale = "ja";
+            } else if (!locale || !(locale === "de" || locale === "en" || locale === "es" || locale === "fr" || locale === "ja")) {
```

Comment:
could be the default `else` statement. Why are we even doing a check here. The logic is - 
* check for known locales
* if found do specific handling
* if not then fallback to `en`